tools/java/common/src/com/google/i18n/phonenumbers/BuildMetadataFromXml.java

```diff
@@ -242,24 +245,27 @@ static boolean loadInternationalFormat(PhoneMetadata.Builder metadata,
       LOGGER.log(Level.SEVERE,
                  "A maximum of one intlFormat pattern for a numberFormat element should be " +
                  "defined.");
-      String countryId = metadata.getId().length() > 0 ?
-          metadata.getId() : Integer.toString(metadata.getCountryCode());
+      String countryId = metadata.id.length() > 0 ?
+          metadata.id : Integer.toString(metadata.countryCode);
       throw new RuntimeException("Invalid number of intlFormat patterns for country: " + countryId);
     } else if (intlFormatPattern.getLength() == 0) {
       // Default to use the same as the national pattern if none is defined.
-      intlFormat.mergeFrom(nationalFormat);
+      intlFormat = PhoneNumberUtil.copyNumberFormat(nationalFormat);
     } else {
-      intlFormat.setPattern(numberFormatElement.getAttribute(PATTERN));
+      intlFormat.pattern = numberFormatElement.getAttribute(PATTERN);
       setLeadingDigitsPatterns(numberFormatElement, intlFormat);
       String intlFormatPatternValue = intlFormatPattern.item(0).getFirstChild().getNodeValue();
       if (!intlFormatPatternValue.equals("NA")) {
-        intlFormat.setFormat(intlFormatPatternValue);
+        intlFormat.format = intlFormatPatternValue;
       }
       hasExplicitIntlFormatDefined = true;
     }
 
-    if (intlFormat.hasFormat()) {
-      metadata.addIntlNumberFormat(intlFormat);
+    if (!intlFormat.format.isEmpty()) {
```

Comment:
The reason I want this to do the check the other way  is because we're conceptually checking for "hazzness", and all places where we do that should be consistent.

Anyway, I don't think this is relevant anymore as I want to encapsulate hazzers into the util class. Let's talk about it.